hooks/useLessonUnlocked.ts

```diff
@@ -0,0 +1,54 @@
+import { useUser, usePathData } from 'hooks'
+import { chapters } from 'content'
+
+function removeAfterHyphen(str: string): string {
+  return str.split('-')[0]
+}
+
+function getIndex(arr: string[], searchStr: string): number {
+  return arr.indexOf(searchStr)
+}
+
+export const useLessonUnlocked = () => {
+  let { lang, pageId, chapterId, lessonId } = usePathData()
+  const userInfo = useUser()
+
+  // if someone vists the website for the first time
+  if (userInfo.user === undefined) {
+    return `/${lang}`
+  } else {
```

Comment:
General rule of thumb for keeping your code clean and easy to understand for other developers that I push alot is to return early (which is what you are doing here), but also to steer away from else blocks if there is no code after that.

```
if (userInfo.user === undefined) {
  return `/${lang}`
}

const userLesson = userInfo.user.progress.lesson
const userChapter = userInfo.user.progress.chapter
const lessonArray = chapters[chapterId].metadata.lessons
const reDirectURL = `/${lang}/${pageId}/${userChapter}/${userLesson}`

if (userChapter.split('-')[1] < chapterId.split('-')[1]) {
  return redirectUrl
}

if (lessonId.includes('intro')) {
  return true
}

if (lessonId.includes('outro')) {
  if (removeAfterHyphen(userLesson) === 'outro') {
    return true
  }
  
  return redirectUrl
}

etc